src/utils.rs

```diff
@@ -143,6 +157,90 @@ pub fn to_human_size(input: U256) -> String {
     tmp.iter().rev().collect()
 }
 
+/// Creates and inserts a given number of empty blocks into the node, with a given interval between them.
+/// The blocks will be empty (contain no transactions).
+/// The test system contracts will be used to force overwriting the block number and timestamp in VM state,
+/// otherwise the VM will reject subsequent blocks as invalid.
+pub fn mine_empty_blocks<S: std::fmt::Debug + ForkSource>(
+    node: &mut InMemoryNodeInner<S>,
+    num_blocks: u64,
+    interval_ms: u64,
+) {
+    // build and insert new blocks
+    for _ in 0..num_blocks {
+        node.current_miniblock = node.current_miniblock.saturating_add(1);
+
+        let block = Block {
+            hash: compute_hash(node.current_miniblock as u32, H256::zero()),
+            number: node.current_batch.into(),
+            timestamp: node.current_timestamp.into(),
+            ..Default::default()
+        };
+
+        node.block_hashes.insert(node.current_miniblock, block.hash);
+        node.blocks.insert(block.hash, block);
+
+        // leave node state ready for next interaction
+        node.current_timestamp = node.current_timestamp.saturating_add(interval_ms);
+        node.current_batch = node.current_batch.saturating_add(1);
+    }
+
+    // roll the vm
+    let (keys, bytecodes) = {
+        let mut storage_view: StorageView<&ForkStorage<S>> = StorageView::new(&node.fork_storage);
+        let mut oracle_tools = OracleTools::new(&mut storage_view, HistoryEnabled);
+
+        let bootloader_code = &node.system_contracts.playground_contracts;
```

Comment:
If we're using `TxExecutionMode::VerifyExecute` below, should we be using the `baseline_contracts` for the bootloader code?

We should also update this to match the new formatting of contract code pulling from `impl SystemContracts`:
```suggestion
        let bootloader_code = inner.system_contracts.contracts(TxExecutionMode::VerifyExecute);
```

Reply:
Ah, I was a bit confused about this- I wanted to use the `unsafeOverrideBlock` method in `SystemContext.sol` so I thought we needed to use playground contract to use this.

I changed it back to `system_contracts` but now it refuses to skip blocks, I get `'The bootloader failed to set previous block hash. Reason: The provided block number is not correct'` when I try to call with a context with `block_number` increased by > 1. When using `playground_contracts` it all seems to work correctly :thinking:

Reply:
Looked closer and we'll definitely run into issues with using the baseline contracts and the enforced sequential block numbers. Let's move forward with the playground contracts 👍